src/back/wgsl/writer.rs

```diff
@@ -96,6 +96,13 @@ impl<W: Write> Writer<W> {
         self.ep_results.clear();
     }
 
+    fn is_builtin_wgsl_struct(&self, ty_struct: &crate::Type) -> bool {
+        ty_struct
+            .name
+            .as_ref()
+            .map_or(false, |name| name.starts_with("__"))
```

Comment:
Can we instead check it's inside `predeclared_types`? It should be more robust.

Reply:
Good idea, done in e8fe048dd3c52064bdcb9db7d8bc39dbeb60f284